### PR TITLE
[#6023] Test explorer should not start Azure emulator when loading tests

### PR DESCRIPTION
**Pull request by sw-joelmut**
Fixes # 6023

## Description
This PR fixes the issue where the `Azure CosmosDb Emulator` was executed in the test discovering process (`Test Explorer`) instead of running when the test is executed, this issue was caused by the `IgnoreOnNoEmulatorFact` attribute.
To address this issue, we refactored the `CosmosDb Emulator` status checker to be included in an [xUnit CollectionFixture](https://xunit.net/docs/shared-context#collection-fixture), ensuring the code to run once, and removing the `IgnoreOnNoEmulatorFact` in favor of `SkippeableFact` from the [Xunit.SkippableFact](https://www.nuget.org/packages/Xunit.SkippableFact) library to skip the tests when the process detects the Emulator isn't running.

## Specific Changes
- Removed the custom `IgnoreOnNoEmulatorFact` attribute, since it was causing the main issue to run the process in discovery time.
- Removed the CosmosDb Emulator status checker from `CosmosDbPartitionStorageFixture` and `CosmosDbStorageTests`.
- Added `CosmosDbFixture` to check if the CosmosDb Emulator is running.
- Added `CosmosDbCollectionDefinition` as part of the [xUnit CollectionFixture](https://xunit.net/docs/shared-context#collection-fixture) process, ensuring the CosmosDb Emulator status checker will run once for `CosmosDbPartitionStorageTests` and `CosmosDbStorageTests` 
  > **Note:** the two test classes will run sequentially due to running them inside the same Collection ([more information](https://xunit.net/docs/running-tests-in-parallel.html#parallelism-in-test-frameworks)).
- Added the [Xunit.SkippableFact](https://www.nuget.org/packages/Xunit.SkippableFact) library, capable of skipping tests based on a condition. Skipping the tests if the Emulator isn't running.
- Updated `CosmosDbPartitionStorageFixture`, `CosmosDbPartitionStorageTests` and `CosmosDbStorageTests` to use the `CosmosDbFixture` and obtain the status of the CosmosDb Emulator.
- Updated the message if the Emulator isn't running to ensure the software is installed and running to execute the tests.

## Testing
The following images show the comparison when the Emulator is or isn't running.
image

### CODE REVIEW

File: tests/Microsoft.Bot.Builder.Azure.Tests/CosmosDbStorageTests.cs [hunk 1]
```diff
@@ -5,7 +5,6 @@
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
 using System.Diagnostics;
-using System.IO;
 using System.Net;
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.Azure.Tests/CosmosDbStorageTests.cs:8]: `using System.Net` is no longer used.

**Comment by sw-joelmut** [comment]
Pipeline run https://dev.azure.com/joelmut93/manx-pipeline-testing/_build/results?buildId=697&view=results
image

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

**Review by NahuelPerales — APPROVED** [review]

File: tests/Microsoft.Bot.Builder.Azure.Tests/CosmosDbStorageTests.cs (hunk 1 above)
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Azure.Tests/CosmosDbStorageTests.cs:8]: Nevermind, it's used in NetCore3.1. Thanks!

**Review by ceciliaavila — APPROVED** [review]

**Comment by sw-joelmut** [comment]
Promoted `https://github.com/microsoft/botbuilder-dotnet/pull/6078`

**Pull request closed without merging**